spras/containers.py

```diff
@@ -161,79 +199,79 @@ def run_container_docker(container: str, command: List[str], volumes: List[Tuple
-                src_dest_map[src_path] = dest
-
-        bind_paths = [f'{prepare_path_docker(src)}:{dest}' for src, dest in volumes]
-
-        out = client.containers.run(container,
-                                    command,
-                                    stderr=True,
-                                    volumes=bind_paths,
-                                    working_dir=working_dir,
-                                    environment=[environment]).decode('utf-8')
-
-        # TODO does this cleanup need to still run even if there was an error in the above run command?
-        # On Unix, files written by the above Docker run command will be owned by root and cannot be modified
-        # outside the container by a non-root user
-        # Reset the file owner and the group inside the container
-        try:
-            # Only available on Unix
-            uid = os.getuid()
-            gid = os.getgid()
-
-            all_modified_volume_contents = set()
-            for src_path in pre_volume_contents.keys():
-                # Assumes the Docker run call is the only process that modified the contents
-                # Only considers files that were added, not files that were modified
-                post_volume_contents = set(src_path.iterdir())
-                modified_volume_contents = post_volume_contents - pre_volume_contents[src_path]
-                modified_volume_contents = [str(convert_docker_path(src_path, src_dest_map[src_path], p)) for p in
-                                            modified_volume_contents]
-                all_modified_volume_contents.update(modified_volume_contents)
-
-            # This command changes the ownership of output files so we don't
-            # get a permissions error when snakemake or the user try to touch the files
-            # Use --recursive because new directories could have been created inside the container
-            # Do not run the command if no files were modified
-            if len(all_modified_volume_contents) > 0:
-                chown_command = ['chown', f'{uid}:{gid}', '--recursive']
-                chown_command.extend(all_modified_volume_contents)
-                chown_command = ' '.join(chown_command)
-                client.containers.run(container,
-                                    chown_command,
-                                    stderr=True,
-                                    volumes=bind_paths,
-                                    working_dir=working_dir,
-                                    environment=[environment]).decode('utf-8')
-
-        # Raised on non-Unix systems
-        except AttributeError:
-            pass
-
-        # TODO: Not sure whether this is needed or where to close the client
-        client.close()
-
     except Exception as err:
-        print(err)
+        err.add_note("An error occurred when fetching the docker daemon: is docker installed and is dockerd running?")
```

Comment:
I'm actually not totally sure of the answer to this... Does SPRAS function with the Docker API from other runtimes like Podman? If so, it may be worth providing a more generic error message, e.g.
```
err.add_note("An error occurred when connecting to the Docker API: is Docker (or a compatible container runtime) installed and running?")
```

At the same time, maybe this is too mountain-dewy and the vast majority of people running SPRAS on their personal machine.

Reply:
Unfortunately, when I was experimenting with Podman support, the docker-py API is not compatible with podman.